### PR TITLE
TypeAlternationTransformer - fix for "array" type in type alternation

### PR DESCRIPTION
**Pull request by kubawerlos**
Fixes https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/5659

### CODE REVIEW

File: src/Tokenizer/Transformer/TypeAlternationTransformer.php [hunk 1]
```diff
@@ -18,7 +18,7 @@
 use PhpCsFixer\Tokenizer\Tokens;
 
 /**
- * Transform `|` operator into CT::T_TYPE_ALTERNATION in `} catch (ExceptionType1 | ExceptionType2 $e) {`.
+ * Transform `|` operator into CT::T_TYPE_ALTERNATION in `function foo(Type1 | Type2 $x) {`.
```
Comment (keradus) [inline src/Tokenizer/Transformer/TypeAlternationTransformer.php:21]: can we keep both examples?

File: tests/Fixer/ClassNotation/VisibilityRequiredFixerTest.php [hunk 2]
```diff
@@ -849,5 +849,13 @@ public function provideFix80Cases()
...
             '<?php class Foo { private int | /* or empty */ null $foo; }',
         ];
+
+        yield [
+            '<?php class Foo { private array|null $foo; }',
+        ];
+
+        yield [
+            '<?php class Foo { private null|array $foo; }',
+        ];
```
Comment (keradus) [inline tests/Fixer/ClassNotation/VisibilityRequiredFixerTest.php:859]: maybe also:

```suggestion
        yield [
            '<?php class Foo { private null|array $foo; }',
        ];

        yield [
            '<?php class Foo { public null|array $foo; }',
            '<?php class Foo { null|array $foo; }',
        ];
```

Comment (kubawerlos) [reply tests/Fixer/ClassNotation/VisibilityRequiredFixerTest.php:859]: `<?php class Foo { null|array $foo; }` is not valid syntax, I've added `static` keyword.

**Review by keradus — APPROVED** [review]

**Comment by keradus** [comment]
Thank you @kubawerlos.

**Pull request merged**